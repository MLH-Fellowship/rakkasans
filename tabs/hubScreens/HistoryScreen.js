import React from "react";
import { View, StyleSheet, Image, TouchableOpacity } from "react-native";
import { useNavigation } from "@react-navigation/native";

import Colors from "../../constants/Colors";
import RectButton from "../../components/RectButton";

export default function HistoryScreen() {
    const navigation = useNavigation();
    return (
        <View style={styles.container}>
            <View style={styles.image}>
                <Image
                    source={require("../../assets/images/history2.png")}
                    style={{ width: 120, height: 120 }}
                />
                <Image
                    source={require("../../assets/images/Flag.jpg")}
                    style={{ width: 170, height: 120, marginLeft: 20 }}
                />
            </View>
            <View style={styles.buttonView}>
                <RectButton
                    text="Notable Events"
                    onPress={() => {
                        navigation.navigate("Notable Events");
                    }}
                    icons={true}
                />
                <RectButton
                    text="MoH Recipients"
                    onPress={() => {
                        navigation.navigate("MoH");
                    }}
                    icons={true}
                />
                <RectButton
                    text="Lineage Honors"
                    onPress={() => {
                        navigation.navigate("Lineage Honors");
                    }}
                    icons={true}
                />
                <RectButton
                    text="Fallen Rakkasans"
                    onPress={() => {
                        navigation.navigate("Fallen Rakkasans");
                    }}
                    icons={true}
                />
                <RectButton
                    text="DMOR_HMOR"
                    onPress={() => {
                      navigation.navigate("DMOR_HMOR");
                    }}
                    icons={true}
                />
                <RectButton
                    text="Division History"
                    onPress={() => {
                        navigation.navigate("Division History");
                    }}
                    icons={true}
                />
                <RectButton text="BN History" onPress={() => {}} icons={true} />
                <RectButton
                    text="3BDE History"
                    onPress={() => {
                        navigation.navigate("38DE History");
                    }}
                    icons={true}
                />
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: Colors.white,
        width: "100%",
        alignItems: "center",
        justifyContent: "space-evenly",
    },
    titleText: {
        color: Colors.primary,
        fontFamily: "andale-mono",
        fontSize: 25,
    },
    buttonView: {
        width: "90%",
    },
    image: {
        flexDirection: "row",
        justifyContent: "space-evenly",
    },
});
