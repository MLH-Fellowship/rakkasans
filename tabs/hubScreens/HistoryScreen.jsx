import React from 'react';
import {
  View, StyleSheet, Image,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';

import Colors from '../../constants/Colors';
import RectButton from '../../components/RectButton';

import HistoryPic from '../../assets/images/history2.png';
import Flag from '../../assets/images/Flag.jpg';

export default function HistoryScreen() {
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: Colors.white,
      width: '100%',
      alignItems: 'center',
      justifyContent: 'space-evenly',
    },
    titleText: {
      color: Colors.primary,

      fontSize: 25,
    },
    buttonView: {
      width: '90%',
    },
    image: {
      flexDirection: 'row',
      justifyContent: 'space-evenly',
    },
  });
  const navigation = useNavigation();
  return (
    <View style={styles.container}>
      <View style={styles.image}>
        <Image
          source={HistoryPic}
          style={{ width: 120, height: 120 }}
        />
        <Image
          source={Flag}
          style={{ width: 170, height: 120, marginLeft: 20 }}
        />
      </View>
      <View style={styles.buttonView}>
        <RectButton
          text="Notable Events"
          onPress={() => {
            navigation.navigate('Notable Events');
          }}
          icons
        />
        <RectButton
          text="MoH Recipients"
          onPress={() => {
            navigation.navigate('MoH');
          }}
          icons
        />
        <RectButton
          text="Lineage Honors"
          onPress={() => {
            navigation.navigate('Lineage Honors');
          }}
          icons
        />
        <RectButton
          text="Fallen Rakkasans"
          onPress={() => {
            navigation.navigate('Fallen Rakkasans');
          }}
          icons
        />
        <RectButton
          text="DMOR_HMOR"
          onPress={() => {
            navigation.navigate('DMOR_HMOR');
          }}
          icons
        />
        <RectButton
          text="Division History"
          onPress={() => {
            navigation.navigate('Division History');
          }}
          icons
        />
        <RectButton text="BN History" onPress={() => {}} icons />
        <RectButton
          text="3BDE History"
          onPress={() => {
            navigation.navigate('38DE History');
          }}
          icons
        />
      </View>
    </View>
  );
}
