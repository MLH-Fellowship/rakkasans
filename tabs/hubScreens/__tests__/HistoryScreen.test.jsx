import React from 'react';
import renderer from 'react-test-renderer';

import HistoryScreen from '../HistoryScreen';

jest.useFakeTimers();
jest.mock('@react-navigation/native');

it('renders correctly', () => {
  const tree = renderer.create(<HistoryScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
