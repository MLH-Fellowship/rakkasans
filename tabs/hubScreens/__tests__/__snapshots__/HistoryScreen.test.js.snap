// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "space-evenly",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 120,
          "width": 120,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 120,
          "marginLeft": 20,
          "width": 170,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginLeft": 30,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        Notable Events
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 30,
            "width": 20,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginLeft": 30,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        MoH Recipients
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 30,
            "width": 20,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginLeft": 30,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        Lineage Honors
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 30,
            "width": 20,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginLeft": 30,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        Fallen Rakkasans
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 30,
            "width": 20,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginLeft": 30,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        DMOR_HMOR
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 30,
            "width": 20,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginLeft": 30,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        Division History
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 30,
            "width": 20,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginLeft": 30,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        BN History
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 30,
            "width": 20,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#020a41",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginLeft": 30,
            "width": 20,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#ffffff",
              "fontSize": 18,
              "marginHorizontal": 30,
              "marginTop": 7,
            },
            Object {
              "fontFamily": "andale-mono",
            },
          ]
        }
      >
        3BDE History
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "marginRight": 30,
            "width": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;
